Mathlib/Algebra/Pointwise/Stabilizer.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Data.Finset.Pointwise
+import Mathlib.GroupTheory.QuotientGroup
+
+/-!
+# Stabilizer of a set under a pointwise action
+
+This file characterises the stabilizer of a set/finset under the pointwise action of a group.
+-/
+
+open Function Set
+open scoped Pointwise
+
+namespace MulAction
+variable {G α : Type*} [Group G] [MulAction G α] {a : G}
+
+section Set
+
+@[to_additive (attr := simp)]
+lemma stabilizer_empty : stabilizer G (∅ : Set α) = ⊤ :=
+  Subgroup.coe_eq_univ.1 $ eq_univ_of_forall fun _a ↦ smul_set_empty
+
+@[to_additive (attr := simp)]
+lemma stabilizer_singleton (b : α) : stabilizer G ({b} : Set α) = stabilizer G b := by ext; simp
+
+@[to_additive]
+lemma mem_stabilizer_set {s : Set α} : a ∈ stabilizer G s ↔ ∀ b, a • b ∈ s ↔ b ∈ s := by
+  refine ⟨fun h b ↦ ?_, fun h ↦ ?_⟩
+  · rw [← (smul_mem_smul_set_iff : a • b ∈ a • s ↔ _), mem_stabilizer_iff.1 h]
+  simp_rw [mem_stabilizer_iff, Set.ext_iff, mem_smul_set_iff_inv_smul_mem]
+  exact ((MulAction.toPerm a).forall_congr' $ by simp [Iff.comm]).1 h
```

Comment:
I imagine rewriting by `mem_stabilizer_iff` before the case split avoids having to rewrite twice.